File: back/src/main/java/com/back/global/jpa/converter/EncryptedStringConverter.java
package com.back.global.jpa.converter;

import com.back.global.util.AesEncryptionUtil;
import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;
import lombok.RequiredArgsConstructor;

@Converter
@RequiredArgsConstructor
public class EncryptedStringConverter implements AttributeConverter<String, String> {

    private final AesEncryptionUtil aesEncryptionUtil;

    @Override
    public String convertToDatabaseColumn(String attribute) {
        if (attribute == null) return null;
        return aesEncryptionUtil.encrypt(attribute);
    }

    @Override
    public String convertToEntityAttribute(String dbData) {
        if (dbData == null) return null;
        return aesEncryptionUtil.decrypt(dbData);
    }
}

File: back/src/main/kotlin/com/back/global/jpa/converter/EncryptedStringConverter.kt
package com.back.global.jpa.converter

import com.back.global.util.AesEncryptionUtil
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter
class EncryptedStringConverter(
    private val aesEncryptionUtil: AesEncryptionUtil
) : AttributeConverter<String?, String?> {

    override fun convertToDatabaseColumn(attribute: String?): String? {
        return attribute?.let { aesEncryptionUtil.encrypt(it) }
    }

    override fun convertToEntityAttribute(dbData: String?): String? {
        return dbData?.let { aesEncryptionUtil.decrypt(it) }
    }
}
